app/controllers/concerns/search_aggregator.rb

```diff
@@ -31,12 +31,16 @@ def aggregate_results(query, results)
 
     all_ontologies = LinkedData::Client::Models::Ontology.all(include: 'acronym,name', include_views: true, display_links: false, display_context: false)
 
+    all_submissions = LinkedData::Client::Models::OntologySubmission.all(include: 'pullLocation', include_views: true, display_links: false, display_context: false)
+
     search_results = grouped_results.map do |group|
       format_search_result(group, all_ontologies)
     end
 
     search_results = merge_sort_federated_results(query, search_results) if federation_enabled?
 
+    search_results = apply_canonical_portal(search_results, all_submissions)
```

Comment:
the canonical choice needs to be done in the function `merge_federated_results`  step as done for the ontologies in the browse page.

Reply:
We need `all_submissions `value only for this function, this is why I didn't want to nest it in for 2 levels.
In all cases the logic here is way different than the one in the browse page.

Reply:
OK